internal/admin/server_121_test.go

```diff
@@ -0,0 +1,18 @@
+//go:build go1.21
+// +build go1.21
```

Comment:
I do feel that those kind of compatibility hacks are hard to maintain (when we upgrade to 1.21+1.22 people may or may not remember to merge this file back into `server_test.go`), so if we expect this list to change frequently between versions, maybe a minimal check is better.

Reply:
It's not terrible, Go treats `+build go1.21` as `>= 1.21`. So it's pretty easy to make it additive.

Reply:
Could we just set the min version to 1.21?

Reply:
That would mean not supporting older compilers. The question is what does this project want to support for Go compiler compatibility.

Personally I recommend following [upstream Go on this](https://go.dev/doc/devel/release#policy).

Reply:
I don't really think we need to merge them back into the list, that just seems like busywork.  This way it's documented which ones showed up in which version, which isn't really useless info even if it's about a version that's outside the support window.